src/dtx/dtx_resync.c

```diff
@@ -196,14 +201,119 @@ dtx_verify_groups(struct ds_pool *pool, struct dtx_memberships *mbs,
 	return true;
 }
 
+int
+dtx_status_handle_one(struct ds_cont_child *cont, struct dtx_entry *dte,
+		      daos_epoch_t epoch, int *tgt_array, int *err)
+{
+	int	rc = 0;
+
+	rc = dtx_check(cont, dte, epoch);
+	switch (rc) {
+	case DTX_ST_COMMITTED:
+	case DTX_ST_COMMITTABLE:
+		/* The DTX has been committed on some remote replica(s),
+		 * let's commit the DTX globally.
+		 */
+		D_GOTO(out, rc = DSHR_NEED_COMMIT);
+	case -DER_INPROGRESS:
+	case -DER_TIMEDOUT:
+		D_WARN("Other participants not sure about whether the "
+		       "DTX "DF_DTI" is committed or not, need retry.\n",
+		       DP_DTI(&dte->dte_xid));
+		D_GOTO(out, rc = DSHR_NEED_RETRY);
+	case DTX_ST_PREPARED: {
```

Comment:
(style)  Possible switch case/default not preceeded by break or fallthrough comment